test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.cs

```diff
@@ -20,238 +20,435 @@
+                var activity = Assert.Single(activities);
+                Assert.Equal(activity.Duration.TotalMilliseconds, sum);
+            }
+            else
+            {
+                Assert.True(sum > 0);
+            }
+
+            var attributes = new Dictionary<string, object>();
+            foreach (var tag in metricPoint.Tags)
+            {
+                attributes[tag.Key] = tag.Value;
+            }
+
+            var expectedAttributeCount = semanticConvention == HttpSemanticConvention.Dupe
+                ? 9 + (tc.ResponseExpected ? 2 : 0)
+                : semanticConvention == HttpSemanticConvention.New
+                    ? 4 + (tc.ResponseExpected ? 1 : 0)
+                    : 5 + (tc.ResponseExpected ? 1 : 0);
+
+            Assert.Equal(expectedAttributeCount, attributes.Count);
+
+            if (semanticConvention == null || semanticConvention.Value.HasFlag(HttpSemanticConvention.Old))
+            {
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpMethod && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeHttpMethod]);
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeNetPeerName && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeNetPeerName]);
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeNetPeerPort && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeNetPeerPort]);
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpScheme && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeHttpScheme]);
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpFlavor && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeHttpFlavor]);
+                if (tc.ResponseExpected)
+                {
+                    Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpStatusCode && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeHttpStatusCode]);
+                }
+                else
+                {
+                    Assert.DoesNotContain(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpStatusCode);
+                }
+            }
+
+            if (semanticConvention != null && semanticConvention.Value.HasFlag(HttpSemanticConvention.New))
+            {
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeHttpRequestMethod && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeHttpMethod]);
+                Assert.Contains(attributes, kvp => kvp.Key == SemanticConventions.AttributeServerAddress && kvp.Value.ToString() == normalizedAttributesTestCase[SemanticConventions.AttributeNetPeerName]);
```

Comment:
Not a blocker for this PR, but this has a bad smell.
Because the json file wasn't updated, it's using the old attribute names from the test case to validate the new attributes.
When we finally ship stable, the json should be updated to have only the New attributes.

Reply:
> When we finally ship stable, the json should be updated to have only the New attributes.

👍